go/arrow/array/concat.go

```diff
@@ -512,6 +518,78 @@ func concat(data []arrow.ArrayData, mem memory.Allocator) (arrow.ArrayData, erro
 		if err != nil {
 			return nil, err
 		}
+	case *arrow.RunLengthEncodedType:
+		physicalLength, overflow := int32(0), false
+		// we can't use gatherChildren because the Offset and Len of
+		// data doesn't correspond to the physical length or offset
+		runs := make([]arrow.ArrayData, len(data))
+		values := make([]arrow.ArrayData, len(data))
+		for i, d := range data {
+			plen := rle.GetPhysicalLength(d)
+			off := rle.GetPhysicalOffset(d)
+
+			runs[i] = NewSliceData(d.Children()[0], int64(off), int64(off+plen))
+			defer runs[i].Release()
+			values[i] = NewSliceData(d.Children()[1], int64(off), int64(off+plen))
+			defer values[i].Release()
+
+			physicalLength, overflow = addOvf32(physicalLength, int32(plen))
+			if overflow {
+				return nil, fmt.Errorf("%w: run length encoded array length must fit into a 32-bit signed integer",
```

Comment:
Does go use 64-bit lengths for arrays?

Reply:
The Go impl uses `int` which will be 32-bit on 32-bit systems and 64-bit on 64-bit systems.